e2e: fleet-default occasionally being deleted during testing

### What steps did you take and what happened?

The e2e long tests are occasionally failing. Failures involve all kind of clusters, this is not an infrastructure problem. The cluster import phase seems to be failing.

Upon further looking in the logs, it seems that the `fleet-default` namespace is deleted, preventing Rancher from importing the cluster:

```txt
2026-08-04T01:19:07.794189381Z 2026/08/04 01:19:07 [WARNING] [mgmt-auth-gr-controller] Namespace fleet-default is terminating in local cluster. Skipping role creation.
2026-08-04T01:19:07.797929745Z 2026/08/04 01:19:07 [INFO] Deleting *v1.Role clusters-create-fleet-default in namespace fleet-default
.
.
.
2026-08-04T01:22:18.340940101Z 2026/08/04 01:22:18 [ERROR] error syncing 'c-8hmjn': handler provisioning-cluster-create: failed to create fleet-default/c-8hmjn provisioning.cattle.io/v1, Kind=Cluster for provisioning-cluster-create c-8hmjn: namespaces "fleet-default" not found, requeuing
```

Reference: https://github.com/rancher/turtles/actions/runs/30866243700/job/91858945823

This seems to be caused by one test using `fleet-default` for deployment, which is later deleted by the spec: https://github.com/rancher/turtles/blob/main/test/e2e/suites/rancher-managed-fleet/rancher_managed_fleet_test.go#L172

### What did you expect to happen?

The `fleet-default` namespace should never be deleted.

### How to reproduce it?

_No response_

### Rancher Turtles version

_No response_

### Anything else you would like to add?

_No response_

### Label(s) to be applied

/kind bug

--- a/test/e2e/helpers.go
+++ b/test/e2e/helpers.go
@@ -87,11 +87,16 @@ func DumpSpecResourcesAndCleanup(ctx context.Context, specName string, clusterPr
 			Namespace:    namespace.Name,
 		}, intervalsGetter(specName, "wait-delete-cluster")...)
 
-		turtlesframework.Byf("Deleting namespace used for hosting the %q test spec", specName)
-		framework.DeleteNamespace(ctx, framework.DeleteNamespaceInput{
-			Deleter: clusterProxy.GetClient(),
-			Name:    namespace.Name,
-		})
+		// Tests that use cloud credentials mapping will be constrained to the shared fleet-default namespace.
+		// Since all CAPI Clusters are imported into fleet-default namespace as well, we need to prevent deletion
+		// of this namespace to not disrupt other tests or other Rancher functionality.
+		if namespace.Name != "fleet-default" {
+			turtlesframework.Byf("Deleting namespace used for hosting the %q test spec", specName)
+			framework.DeleteNamespace(ctx, framework.DeleteNamespaceInput{
+				Deleter: clusterProxy.GetClient(),
+				Name:    namespace.Name,
+			})
+		}
 	}
 	cancelWatches()
 }

